Stub for argument of function is not generated in test file

## Steps to reproduce

1. Open and configure project `coreutils`
2. Generate tests for function `mpsort_into_tmp` in `gnulib/lib/mpsort.c`
3. Run tests
4. See logs

```
clang-10: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [/home/utbot/projects/coreutils/build/lib/liblibcoreutils.a.so] Error 1
make: *** Waiting for unfinished jobs....
/home/utbot/projects/coreutils/utbot_tests/gnulib/lib/mpsort_test.cpp:30:87: error: use of undeclared identifier '_mpsort_into_tmp_cmp_stub'
    mpsort_into_tmp((const void **) base, 6477909508830039056UL, (const void **) tmp, _mpsort_into_tmp_cmp_stub);
                                                                                      ^
/home/utbot/projects/coreutils/utbot_tests/gnulib/lib/mpsort_test.cpp:47:69: error: use of undeclared identifier '_mpsort_into_tmp_cmp_stub'
    mpsort_into_tmp((const void **) base, 4UL, (const void **) tmp, _mpsort_into_tmp_cmp_stub);
                                                                    ^
/home/utbot/projects/coreutils/utbot_tests/gnulib/lib/mpsort_test.cpp:64:69: error: use of undeclared identifier '_mpsort_into_tmp_cmp_stub'
    mpsort_into_tmp((const void **) base, 5UL, (const void **) tmp, _mpsort_into_tmp_cmp_stub);
                                                                    ^
3 errors generated.
```

--- a/server/CMakeLists.txt
+++ b/server/CMakeLists.txt
@@ -178,7 +178,7 @@ if (ENABLE_UNIT_TESTS)
     target_link_libraries(
             UTBot_UnitTests
             PUBLIC
-            gtest_main
+            gtest
             UnitTestBotLib
     )
 

--- a/server/src/Tests.cpp
+++ b/server/src/Tests.cpp
@@ -591,7 +591,7 @@ void KTestObjectParser::assignTypeUnnamedVar(Tests::MethodTestCase &testCase,
         }
     }
     {
-        int ind = findObjectIndex(testCase.objects, KLEERESULT);
+        int ind = findObjectIndex(testCase.objects, KleeUtils::RESULT_VARIABLE_NAME);
         if (ind != testCase.objects.size()) {
             visited[ind] = true;
             order.push({ind, methodDescription.returnType});

--- a/server/src/Tests.h
+++ b/server/src/Tests.h
@@ -29,7 +29,6 @@ namespace tests {
     using std::shared_ptr;
 
     const std::string LAZYNAME = "unnamed";
-    const std::string KLEERESULT = "utbot_result";
 
     using MapAddressName = std::unordered_map<size_t , std::string>;
 
@@ -396,6 +395,7 @@ namespace tests {
             std::optional<MethodParam> classObj;
             std::string name;
             typedef std::unordered_map<string, string> SuiteNameToCodeTextMap;
+            std::string stubsText;
             SuiteNameToCodeTextMap codeText;
             std::string paramsString;
 

--- a/server/src/printers/Printer.cpp
+++ b/server/src/printers/Printer.cpp
@@ -645,6 +645,15 @@ namespace printer {
         ss << tests.stubs << NL;
     }
 
+    void Printer::writeStubsForParameters(const Tests &tests) {
+        for (const auto &[methodName, methodDescription] : tests.methods) {
+            if (methodDescription.stubsText.empty()) {
+                continue;
+            }
+            ss << methodDescription.stubsText << NL;
+        }
+    }
+
     utbot::Language Printer::getLanguage() const {
         return srcLanguage;
     }

--- a/server/src/printers/Printer.h
+++ b/server/src/printers/Printer.h
@@ -215,6 +215,8 @@ namespace printer {
 
         void writeStubsForStructureFields(const Tests &tests);
 
+        void writeStubsForParameters(const Tests &tests);
+
         void writeStubForParam(const types::TypesHandler* typesHandler,
                                const std::shared_ptr<types::FunctionInfo> &fInfo,
                                const string& name,

--- a/server/src/printers/TestsPrinter.cpp
+++ b/server/src/printers/TestsPrinter.cpp
@@ -57,6 +57,9 @@ void TestsPrinter::joinToFinalCode(Tests &tests, const fs::path& generatedHeader
     }
     writeStubsForStructureFields(tests);
     ss << NL;
+    writeStubsForParameters(tests);
+    ss << NL;
+
     tests.regressionMethodsNumber = printSuiteAndReturnMethodsCount(Tests::DEFAULT_SUITE_NAME, tests.methods);
     tests.errorMethodsNumber = printSuiteAndReturnMethodsCount(Tests::ERROR_SUITE_NAME, tests.methods);
     ss << RB();
@@ -95,6 +98,9 @@ void TestsPrinter::genCode(Tests::MethodDescription &methodDescription,
 
     writeStubsForFunctionParams(typesHandler, methodDescription, false);
     writeExternForSymbolicStubs(methodDescription);
+
+    methodDescription.stubsText = ss.str();
+    resetStream();
 
     genCodeBySuiteName(Tests::DEFAULT_SUITE_NAME,
                        methodDescription,

--- a/server/test/framework/Regression_Tests.cpp
+++ b/server/test/framework/Regression_Tests.cpp
@@ -219,4 +219,19 @@ namespace {
                 } }),
             "libbpf_set_print");
     }
+
+    TEST_F(Regression_Test, Unused_Function_Pointer_Parameter) {
+        fs::path source = getTestFilePath("PR153.c");
+        auto [testGen, status] = createTestForFunction(source, 9);
+
+        ASSERT_TRUE(status.ok()) << status.error_message();
+
+        checkTestCasePredicates(
+            testGen.tests.at(source).methods.begin().value().testCases,
+            vector<TestCasePredicate>(
+                { [](const tests::Tests::MethodTestCase &testCase) {
+                    return testCase.isError();
+                } }),
+            "unused");
+    }
 }

--- a/server/test/suites/regression/CMakeLists.txt
+++ b/server/test/suites/regression/CMakeLists.txt
@@ -30,4 +30,6 @@ add_library(PR123 PR123.c)
 
 add_library(PR124 PR124.c)
 
+add_library(PR153 PR153.c)
+
 set_target_properties(regression PROPERTIES LINK_WHAT_YOU_USE TRUE)

--- a/server/test/suites/regression/PR153.c
+++ b/server/test/suites/regression/PR153.c
@@ -1,0 +1,10 @@
+/*
+ * Copyright (c) Huawei Technologies Co., Ltd. 2012-2021. All rights reserved.
+ */
+
+#include <assert.h>
+
+typedef int (*comparison_function)(void const *, void const *);
+void unused(comparison_function cmp) {
+    assert(0);
+}

